stl/inc/chrono

```diff
@@ -5529,8 +5599,25 @@ struct _Chrono_formatter {
         _THROW(format_error("Incompatible modifier for type"));
     }
 
+    template <class _Ty>
+    _NODISCARD constexpr bool _Is_valid_type(const char _Type) noexcept {
+        if constexpr (is_same_v<_Ty, _CHRONO day>) {
+            return _Type == 'd' || _Type == 'e';
+        } else if constexpr (is_same_v<_Ty, _CHRONO month>) {
+            return _Type == 'b' || _Type == 'B' || _Type == 'h' || _Type == 'm';
+        } else if constexpr (is_same_v<_Ty, _CHRONO year>) {
+            return _Type == 'Y' || _Type == 'y' || _Type == 'C';
+        } else if constexpr (is_same_v<_Ty, _CHRONO year_month_day>) {
+            return _Type == 'D' || _Type == 'F' || _Is_valid_type<_CHRONO year>(_Type)
+                || _Is_valid_type<_CHRONO month>(_Type) || _Is_valid_type<_CHRONO day>(_Type);

```

Comment:
I'm wondering if the specifiers that are valid for `weekday` are also valid here since a specific year, month, and weekday would give us a particular weekday.  Similarly, would `'U'` and `'V'` be valid specifiers since we could get a particular week from a `year_month_day`?

Reply:
We do not have a weekday. I based which specifiers are legal from [put_time](ttps://en.cppreference.com/w/cpp/io/manip/put_time), and since it isn't trivial to compute the weekday from just a year/month/day, I think it doesn't apply. I could be very wrong though.

Reply:
After discussing on discord, I agree it makes sense to be able to derive some more information from a `year_month_day` than I originally implemented.